### PR TITLE
Document actix-web-codegen

### LINKED ISSUE #105: Add documentation about actix-web-codegen
So the codegen part is great. I mean this syntax: https://github.com/actix/examples/blob/44053504ad39b2bb023712c7577a06091629ad41/basics/src/main.rs#L24

I think it makes a lot of sense to document that as it makes actix-web easier to read and more approachable in general.

### PR DESCRIPTION
**Pull request by svenstaro**
This fixes #105

### CODE REVIEW

File: content/docs/getting-started.md [hunk 1]
```diff
@@ -43,11 +43,30 @@ accepts a function that should return an application factory.
 That's it! Now, compile and run the program with `cargo run`.
 Head over to ``http://localhost:8088/`` to see the results.
 
+### Alternative syntax
```
Comment (vmalloc) [inline content/docs/getting-started.md]: As a title, I would prefer "Using Attribute Macros to Define Routes" as a title for this. Readers shouldn't be interested in which crate exactly implements this...

File: content/docs/getting-started.md [hunk 2]
```diff
@@ -43,11 +43,30 @@ accepts a function that should return an application factory.
...
+
+You might prefer the alternative syntax provided by [actix-web-codegen] which
```
Comment (vmalloc) [inline content/docs/getting-started.md]: I would start this with "It is also possible to define routes using macro attributes".
In any case "might" shouldn't be used here

File: content/docs/getting-started.md [hunk 3]
```diff
@@ -43,11 +43,30 @@ accepts a function that should return an application factory.
...
+allows you to specify the routes above your functions like so:
+
+{{< include-example example="getting-started" section="alternative">}}
+
+You can then declare this function using `service()`:
```
Comment (vmalloc) [inline content/docs/getting-started.md]: register your route/service, not define this function

File: content/docs/getting-started.md [hunk 4]
```diff
@@ -43,11 +43,30 @@ accepts a function that should return an application factory.
...
+
+```rust
+App::new()
+    .service(index3)
+```
+
+For consistency reasons, this documentation only uses the regular syntax shown at the
```
Comment (vmalloc) [inline content/docs/getting-started.md]: regular --> explicit syntax IMO

File: examples/getting-started/src/main.rs [hunk 5]
```diff
@@ -10,6 +10,15 @@ fn index2() -> impl Responder {
 }
 // </setup>
 
+// <alternative>
+use actix_web::get;
+
+#[get("/alternative")]
```
Comment (vmalloc) [inline examples/getting-started/src/main.rs]: avoid "alternative" here - it should be the same path as the previous example to demonstrate equivalence IMO

**Comment by vmalloc** [comment]
Thanks for the PR! I think some small fixes are needed - commented on the changes directly

**Comment by svenstaro** [comment]
Great points overall. The only thing I didn't do as you said is having the same route path. I think it's confusing for users if they want to just copy that into their code as then the paths would overlap. However, I added a path that's more in line with the current paths shown in the examples so that it should be an easy concept to grasp.

File: content/docs/getting-started.md [hunk 6]
```diff
@@ -43,24 +43,26 @@ accepts a function that should return an application factory.
 That's it! Now, compile and run the program with `cargo run`.
 Head over to ``http://localhost:8088/`` to see the results.
 
-### Alternative syntax
+### Using Attribute Macros to Define Routes
 
-You might prefer the alternative syntax provided by [actix-web-codegen] which
+Alternatively, you can define routes using macro attributes which
 allows you to specify the routes above your functions like so:
```
Comment (vmalloc) [inline content/docs/getting-started.md]: `allow` instead of `allows`

**Pull request merged**